packages/toolkit/src/query/react/ApiProvider.tsx

```diff
@@ -45,7 +45,13 @@ export function ApiProvider<A extends Api<any, {}, any, any>>(props: {
     })
   )
   // Adds the event listeners for online/offline/focus/etc
-  setupListeners(store.dispatch, props.setupListeners)
+  useEffect(
+    (): undefined | (() => void) =>
+      props.setupListeners === false
+        ? undefined
+        : setupListeners(store.dispatch, props.setupListeners),
+    [props.setupListeners]
+  )
```

Comment:
Hmmm, do we need to rework `setupListeners` and `initialized` make sure we're cleaning up the subscriptions here if we're putting it in an effect?

Reply:
Actually, that's a good point. Especially with React 18 double-running effects in dev.

Reply:
`setupListeners` returns an `unsubscribe` function so this should be fine as it is.